packages/block-library/src/social-link/edit.native.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * External dependencies
+ */
+import { View, Animated, Easing, TouchableWithoutFeedback } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { BlockControls } from '@wordpress/block-editor';
+import { useEffect, useState, useRef } from '@wordpress/element';
+import {
+	ToolbarGroup,
+	ToolbarButton,
+	LinkSettings,
+} from '@wordpress/components';
+import { usePreferredColorSchemeStyle } from '@wordpress/compose';
+import { __, sprintf } from '@wordpress/i18n';
+import { link, Icon } from '@wordpress/icons';
+/**
+ * Internal dependencies
+ */
+import { getIconBySite, getNameBySite } from './social-list';
+import styles from './editor.scss';
+
+const ANIMATION_DELAY = 300;
+const ANIMATION_DURATION = 400;
+
+const SocialLinkEdit = ( {
+	attributes,
+	setAttributes,
+	isSelected,
+	onFocus,
+} ) => {
+	const { url, service } = attributes;
+	const [ isLinkSheetVisible, setIsLinkSheetVisible ] = useState( false );
+	const [ isValue, setIsValue ] = useState( !! url );
+
+	const activeIcon =
+		styles[ `wp-social-link-${ service }` ] || styles[ `wp-social-link` ];
+
+	const inactiveIcon = usePreferredColorSchemeStyle(
+		styles.inactiveIcon,
+		styles.inactiveIconDark
+	);
+
+	const animatedValue = useRef( new Animated.Value( 0 ) ).current;
+
+	const IconComponent = getIconBySite( service )();
+	const socialLinkName = getNameBySite( service );
+
+	// When new social icon is added link sheet is opened automatically
+	useEffect( () => {
+		if ( isSelected && ! url ) {
+			setIsLinkSheetVisible( true );
+		}
+	}, [] );
+
+	useEffect( () => {
+		if ( ! url ) {
+			setIsValue( false );
+			animatedValue.setValue( 0 );
+		} else if ( url ) {
+			animateColors();
+		}
+	}, [ url ] );
+
+	const interpolationColors = {
+		backgroundColor: animatedValue.interpolate( {
+			inputRange: [ 0, 1 ],
+			outputRange: [
+				inactiveIcon.backgroundColor,
+				activeIcon.backgroundColor,
+			],
+		} ),
+		color: animatedValue.interpolate( {
+			inputRange: [ 0, 1 ],
+			outputRange: [ inactiveIcon.color, activeIcon.color ],
+		} ),
+		stroke: '',
+	};
+
+	const { backgroundColor, color, stroke } = isValue
+		? activeIcon
+		: interpolationColors;
+
+	const options = {
```

Comment:
It seems like something that could be a const declared outside of the component.